### PR TITLE
9 home card function remove and edit

### PR DESCRIPTION
**Pull request by alexanderstrigun**
**key achievements**

- individual gift items can now be removed via clicking the delete icon
- individual gift items can no be edited via clicking the edit icon
- in the dropdown component, the custom created dropdown was replaced with a select element

### CODE REVIEW

File: components/AddOverlay.jsx [hunk 1]
```diff
@@ -18,10 +18,16 @@ export const AddOverlay = ({
...
   priceInputHandler,
   giftListHandler,
+  selectedGiftHandler,
+  giftListEditor,
 }) => {
   const clickHandlerCloseButton = (event) => {
     event.preventDefault();
     overlayHandler();
+    selectedGiftHandler("", "", "", "", "");
+    if (editMode === true) {
```
Comment (StefanBehring) [inline components/AddOverlay.jsx:28]:
```suggestion
    if (editMode) {
```
Does the same

File: components/AddOverlay.jsx [hunk 2]
```diff
@@ -41,11 +47,18 @@ export const AddOverlay = ({
...
   return (
     <>
       {overlayState ? (
         <FixedPositionDiv>
-          <Form onSubmit={submitHandlerForm}>
+          <Form
+            onSubmit={editMode ? submitHandlerFormEditMode : submitHandlerForm}
+          >
             <Button onClick={clickHandlerCloseButton}>
               <GrClose></GrClose>
```
Comment (StefanBehring) [inline components/AddOverlay.jsx:63]:
```suggestion
              <GrClose />
```

File: components/Card.jsx [hunk 3]
```diff
@@ -2,15 +2,45 @@ import styled from "styled-components";
...
-export const Card = ({ giftList }) => {
+export const Card = ({
+  giftList,
+  giftListRemover,
+  selectedGiftHandler,
+  overlayHandler,
+  editModeHandler,
+}) => {
   const gift = giftList.map((individualGift) => {
+    const clickHandlerRemoveGift = () => {
```
Comment (djfarly) [inline components/Card.jsx:13]: `handleRemoveGift`

File: components/Card.jsx [hunk 4]
```diff
@@ -2,15 +2,45 @@ import styled from "styled-components";
...
+      let giftID = individualGift.id;
+      let selectedGiftIndex = giftList.findIndex((element) => {
+        return element.id === giftID;
+      });
```
Comment (djfarly) [inline components/Card.jsx:17]: This seems redundant. `index` is the second parameter of `giftList.map((individualGift, index) => {`. Here you're doing extra work that just gives you something you already know. `selectedGiftIndex` is the same as `index`.

File: components/Card.jsx [hunk 5]
```diff
@@ -2,15 +2,45 @@ import styled from "styled-components";
...
+      giftListRemover(selectedGiftIndex);
+    };
+
+    const clickHandlerEditGift = () => {
```
Comment (djfarly) [inline components/Card.jsx:21]: phew - maybe we can talk about naming on monday? I'll stop here - i think you got the gist 😄🥺

File: components/Card.jsx [hunk 6]
```diff
@@ -2,15 +2,45 @@ import styled from "styled-components";
...
+      selectedGiftHandler(
+        individualGift.id,
+        individualGift.name,
+        individualGift.category,
+        individualGift.link,
+        individualGift.price
+      );
```
Comment (djfarly) [inline components/Card.jsx:28]: Consider using a single object parameter instead of multiple params.

File: components/Dropdown.jsx [hunk 7]
```diff
@@ -1,65 +1,19 @@
...
-  dropdownInputHandler,
-  categoryDD,
-}) => {
-  const clickHandlerDropdown = (event) => {
-    event.preventDefault();
-    dropdownHandler();
-  };
-
+export const Dropdown = ({ dropdownInputHandler, categoryDD }) => {
   const categoryDDLiItems = categoryDD.map((categoryDDitem) => {
```
Comment (djfarly) [inline components/Dropdown.jsx:2]: Why is `categoryDD` an array?

File: pages/_app.js [hunk 8]
```diff
@@ -34,20 +31,18 @@ function CustomApp({ Component, pageProps }) {
     setDropdownInput(dropdownInput);
   };
   ///////////////END//////////////////
-
+  console.log(dropdownInput);
```
Comment (djfarly) [inline pages/_app.js:34]: Remove console.log

File: pages/_app.js [hunk 9]
```diff
@@ -67,8 +62,60 @@ function CustomApp({ Component, pageProps }) {
...
+    category: "",
+    link: "",
+    price: "",
+  });
+  ////////////the inputs in the form shall change when the user clicks the edit button
+  useEffect(() => {
+    nameInputHandler(selectedGift.name);
+    dropdownInputHandler(selectedGift.category);
+    linkInputHandler(selectedGift.link);
+    priceInputHandler(selectedGift.price);
```
Comment (djfarly) [inline pages/_app.js:86]: I am confused as to what these handlers do…

**Review by djfarly — CHANGES_REQUESTED** [review]
Hej Alex,

thanks for the PR! Great progress.

I noticed a few things regarding naming as well as function signatures. Maybe we should talk about namings on monday - i think that would be a bit easier than me typing it all out.

Anyway keep going strong! ✌️

**Comment by alexanderstrigun** [comment]
added changes as suggested by jan.

**Comment by alexanderstrigun** [comment]
removed a console log

File: components/AddOverlay.jsx (hunk 2 above)
Comment (alexanderstrigun) [reply components/AddOverlay.jsx]: thx for the hint! :)

File: server/server.mjs [hunk 10]
```diff
@@ -0,0 +1,23 @@
...
+    default:
+      break;
+  }
+
+  console.log({ url, method });
+  res.writeHead(200, { "Content-type": "application/json" });
+  res.end(JSON.stringify({ data: "Hello World" }));
+});
+
+server.listen(8000);
```
Comment (djfarly) [inline server/server.mjs:23]: I think you do not need this file.

For all other things: LGTM 😱✌️

**Review by djfarly — APPROVED** [review]
🎊

File: server/server.mjs (hunk 10 above)
Comment (alexanderstrigun) [reply server/server.mjs:23]: cool, thanks!

**Pull request merged**